src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/General/Helpers.cs

```diff
@@ -5,16 +5,18 @@
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
 using System.Text;
 
 namespace System.Reflection.TypeLoading
 {
     internal static class Helpers
     {
+        [return: NotNullIfNotNull("original")]
         public static T[] CloneArray<T>(this T[] original)
         {
             if (original == null)
-                return null;
+                return null!;
```

Comment:
This method should accept and return nullable, then the attribute ` [return: NotNullIfNotNull("original")]` will be effective too